### PR TITLE
bugfix/LEAF-1343-sanitize-qr - validate and scrub QR payloads

### PR DESCRIPTION
**Pull request by mgaoVA**

### CODE REVIEW

**Comment by dastan-rahimi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VALEAF/LEAF/pullRequest?prid=5169795&bid=16971192)

File: libs/qrcode/index.php [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
+    $matches = array();
+    switch($_SERVER['HTTP_HOST']) {
+        case 'leaf.va.gov':
+            $HTTP_HOST = 'leaf.va.gov';
+            break;
+        case 'leaf-preprod.va.gov':
+            $HTTP_HOST = 'leaf-preprod.va.gov';
+            break;
+        case 'leaf-dev.va.gov':
+            $HTTP_HOST = 'leaf-dev.va.gov';
```
Comment (dastan-rahimi) [inline libs/qrcode/index.php]: We can't have these domain names in a public repo. We can pull in globals.php and compare HTTP_HOST instead

Comment (mgaoVA) [reply libs/qrcode/index.php]: Updated. Note that the update will potentially break QR-related functionality for some sites, but it's more important to get this fix merged. The breaking functionality is accepted.

**Comment by dastan-rahimi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VALEAF/LEAF/pullRequest?prid=5169795&bid=16971192)

**Comment by dastan-rahimi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VALEAF/LEAF/pullRequest?prid=5169795&bid=16971192)

**Comment by dastan-rahimi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VALEAF/LEAF/pullRequest?prid=5169795&bid=16971192)

**Pull request merged**